src/System.Linq/tests/AggregateTests.cs

```diff
@@ -0,0 +1,148 @@
+                    where !String.IsNullOrEmpty(x)
+                    select x;
+
+            Assert.Equal(q.Aggregate((x, y) => x + y), q.Aggregate((x, y) => x + y));
+        }
+
+        [Fact]
+        public void EmptySource()
+        {
+            int[] source = { };
+            
+            Assert.Throws<InvalidOperationException>(() => source.Aggregate((x, y) => x + y));
+        }
+
+        [Fact]
+        public void SingleElement()
+        {
+            int[] source = { 5 };
+            int expected = 5;
+
+            Assert.Equal(expected, source.Aggregate((x, y) => x + y));
+
+        }
+
+        [Fact]
+        public void TwoElements()
+        {
+            int[] source = { 5, 6 };
+            int expected = 11;
+
+            Assert.Equal(expected, source.Aggregate((x, y) => x + y));
+        }
+
+        [Fact]
+        public void MultipleElements()
+        {
+            int[] source = { 5, 6, 0, -4 };
+            int expected = 7;
+
+            Assert.Equal(expected, source.Aggregate((x, y) => x + y));
+        }
+
+        [Fact]
+        public void EmptySourceAndSeed()
+        {
+            int[] source = { };
+            long seed = 2;
+            long expected = 2;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y));
+        }
+
+        [Fact]
+        public void SingleElementAndSeed()
+        {
+            int[] source = { 5 };
+            long seed = 2;
+            long expected = 10;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y));
+        }
+
+        [Fact]
+        public void TwoElementsAndSeed()
+        {
+            int[] source = { 5, 6 };
+            long seed = 2;
+            long expected = 60;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y));
+        }
+
+        [Fact]
+        public void MultipleElementsAndSeed()
+        {
+            int[] source = { 5, 6, 2, -4 };
+            long seed = 2;
+            long expected = -480;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y));
+        }
+
+        [Fact]
+        public void NoElementsSeedResultSeletor()
+        {
+            int[] source = { };
+            long seed = 2;
+            double expected = 7;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y, x => x + 5.0));
+        }
+
+        [Fact]
+        public void SingleElementSeedResultSelector()
+        {
+            int[] source = { 5 };
+            long seed = 2;
+            long expected = 15;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y, x => x + 5.0));
+        }
+
+        [Fact]
+        public void TwoElementsSeedResultSelector()
+        {
+            int[] source = { 5, 6 };
+            long seed = 2;
+            long expected = 65;
+
+            Assert.Equal(expected, source.Aggregate(seed, (x, y) => x * y, x => x + 5.0));
```

Comment:
I notice that the original test had ```x + 5``` and you changed it to ```x + 5.0```... any reason?  Same for the next one.